### ISSUE TITLE
[BUG] "[vite]: Rollup failed to resolve import" with vite-tsconfig-paths plugin

### ISSUE THREAD

**Issue by Giorno-Giovana** [microsoft/playwright#29196]
### System info
- Playwright Version: playwright/experimental-ct-react@1.41.1
- Operating System: All
- Browser: All

### Source code

https://codesandbox.io/p/devbox/playwright-vite-representation-hxndcz

### Problem
```
[vite]: Rollup failed to resolve import "@/components/Counter" from "/workspaces/workspace/playwright/index.tsx".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
Error: [vite]: Rollup failed to resolve import "@/components/Counter" from "/workspaces/workspace/playwright/index.tsx".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
```

The reason is my `vite-tsconfig-paths` plugin. My App is working fine and TS imports too. But, for some reason, not in playwright.

**Steps to reproduce**
- Run `npm run test-ct`

**Comment by yury-s** [comment]
Can you try changing your playwright config like this:

```js
import react from "@vitejs/plugin-react";
...
    ctViteConfig: {
      plugins: [tsconfigPaths(), react()],
    },

```
Would that work?

**Comment by Giorno-Giovana** [comment]
No, it didn't help

**Comment by Giorno-Giovana** [comment]
I tried to debug using `DEBUG=vite-tsconfig-paths` to compare what is the difference between `npm run dev` and `npm run-ct`. The log turned out to be absolutely identical

```
  vite-tsconfig-paths options.root   == undefined +0ms
  vite-tsconfig-paths project root   == /workspaces/workspace +0ms
  vite-tsconfig-paths workspace root == /workspaces/workspace +0ms
  vite-tsconfig-paths projects: [ '/workspaces/workspace/tsconfig.json' ] +15ms
  vite-tsconfig-paths config loaded: {
  configPath: '/workspaces/workspace/tsconfig.node.json',
  config: {
    compilerOptions: {
      composite: true,
      skipLibCheck: true,
      module: 'ESNext',
      moduleResolution: 'bundler',
      allowSyntheticDefaultImports: true
    },
    include: [ 'vite.config.ts' ]
  }
} +4ms
  vite-tsconfig-paths [!] missing baseUrl and paths: "/workspaces/workspace/tsconfig.node.json" +0ms
  vite-tsconfig-paths config loaded: {
  configPath: '/workspaces/workspace/tsconfig.json',
  config: {
    compilerOptions: {
      target: 'ES2020',
      useDefineForClassFields: true,
      lib: [ 'ES2020', 'DOM', 'DOM.Iterable' ],
      module: 'ESNext',
      skipLibCheck: true,
      moduleResolution: 'bundler',
      allowImportingTsExtensions: true,
      resolveJsonModule: true,
      isolatedModules: true,
      noEmit: true,
      jsx: 'react-jsx',
      strict: true,
      noUnusedLocals: true,
      noUnusedParameters: true,
      noFallthroughCasesInSwitch: true,
      paths: { '@/*': [ 'src/*' ] },
      baseUrl: '/workspaces/workspace'
    },
    include: [ 'src' ],
    references: [ { path: './tsconfig.node.json' } ]
  }
} +0ms
  vite-tsconfig-paths compiled globs: {
  includers: [ /^\.\/src\/((?:[^/]*(?:\/|$))*)$/ ],
  excluders: [
    /^\.\/((?:[^/]*(?:\/|$))*)node_modules\/((?:[^/]*(?:\/|$))*)$/,
    /^\.\/((?:[^/]*(?:\/|$))*)bower_components\/((?:[^/]*(?:\/|$))*)$/,
    /^\.\/((?:[^/]*(?:\/|$))*)jspm_packages\/((?:[^/]*(?:\/|$))*)$/
  ]
} +1ms
```

**Comment by Giorno-Giovana** [comment]
If I write this code in playwright config, then everything will work. But this does not negate the fact that for some reason playwright does not want to work with vite-tsconfig-paths

```ts
alias: {
  '@': resolve(dirname('.'), './src'),
},
```

**Comment by pavelfeldman** [comment]
This should be fixed in 1.42
